File: abc_training/C_diff/C_MonsterBattleRoyale.cpp
//        C - Monsters Battle Royale
// ----------------------------------------
// 問題
// https://atcoder.jp/contests/abc118/tasks/abc118_c
// ----------------------------------------

// segment fault

#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

int main() {
    ll N; cin >> N;
    set<ll> vals;
    for (int i = 0; i < N; i++) {
        ll val; cin >> val;
        vals.insert(val);
    }

    // 最小の値で貪欲に剰余をとっていく、0となった値は除く
    ll mini;
    do {
        // print
        for (auto itr = vals.begin(); itr != vals.end(); itr++) {
            cerr << *itr << " " << endl;
        }
        cerr << endl;

        mini = *vals.begin();
        vals.erase(vals.begin());
        for (auto itr = vals.begin(); itr != vals.end(); itr++) {
            vals.insert(*itr % mini);
            vals.erase(itr);
        }
    } while (!vals.empty());

    cout << mini << endl;
}

File: abc_training/C/C_MonsterBattleRoyale.py
#       C - Monsters Battle Royale
# ----------------------------------------
# 問題
# https://atcoder.jp/contests/abc118/tasks/abc118_c

# 焦らない

# AC
# ----------------------------------------

# 計算量は　O(n)

import heapq

if __name__ == "__main__":
    N = int(input())
    A = list(map(int, input().split()))
    heapq.heapify(A)

    while len(A) > 1:
        mini = heapq.heappop(A)

        while A and A[0] > mini:
            val = heapq.heappop(A) % mini
            if val != 0:
                heapq.heappush(A, val)

        if A and A[0] != mini:
            heapq.heappush(A, mini)
    
    print(mini)
